frontend/model/contracts/group.js

```diff
@@ -261,38 +261,79 @@ function updateGroupStreaks ({ state, getters }) {
   }
 }
 
+const removeGroupChatroomProfile = (state, chatRoomID, member) => {
+  console.log('jjjj removeGroupChatroomProfile', JSON.parse(JSON.stringify({ state, chatRoomID, member })))
+  Vue.set(state.chatRooms[chatRoomID], 'users',
+    Object.fromEntries(
+      Object.entries(state.chatRooms[chatRoomID].users)
+        .map(([memberKey, profile]) => {
+          console.log('jjjj removeGroupChatroomProfile ii', {
+            memberKey, member, profile, active: (profile: any)?.status === PROFILE_STATUS.ACTIVE
+          })
+          if (memberKey === member && (profile: any)?.status === PROFILE_STATUS.ACTIVE) {
+            return [memberKey, { status: PROFILE_STATUS.REMOVED }]
```

Comment:
This line seems to be removing many profile fields... that seems wrong.

Reply:
There isn't much in the profile _and_ this is for removed members. Would you prefer `{ ...profile, status: PROFILE_STATUS.REMOVED }` instead?

Reply:
Yes, I think that would be much more robust.